lib/optimizely_server_side/helpers/support.rb

```diff
@@ -24,17 +24,41 @@ module Support
     def experiment(experiment_key, options = {}, &blk)
       # Merge any other options to user attributes
       OptimizelyServerSide.configuration.user_attributes.merge!(options.stringify_keys)
-      variation_key = optimizely_sdk_project_instance(experiment_key)
+      variation_key = optimizely_sdk_project_instance(experiment_key, "experiment")
       OptimizelyServerSide::Experiment.new(experiment_key, variation_key).start(&blk)
     end
     alias_method :feature_flip, :experiment
 
-    def optimizely_sdk_project_instance(experiment_key)
+    def optimizely_sdk_project_instance(experiment_key, type)
+      case type
+      when "experiment"
+        get_optimizely_instance.activate(experiment_key, visitor_id,
+                OptimizelyServerSide.configuration.user_attributes.reject { |k,v| k == 'visitor_id'})
+      when "feature"
+        get_optimizely_instance.is_feature_enabled(experiment_key, visitor_id,
+                OptimizelyServerSide.configuration.user_attributes.reject { |k,v| k == 'visitor_id'})
```

Comment:
As you have extracted `visitor_id` can we also have `user_attributes` ?

```ruby
def user_attributes
  OptimizelyServerSide.configuration.user_attributes.reject { |k,v| k == 'visitor_id'})
end
```

Reply:
👍 Will do